tests/php/Forms/NumericFieldTest.php

```diff
@@ -725,6 +725,14 @@ public function testSetSubmittedValue(
         $this->assertSame($cleanedInput, $field->getFormattedValue(), 'getFormattedValue()');
     }
 
+    public function testSetSubmittedValueNonString(): void
```

Comment:
Should add the the dataprovider for testSetValue() instead of creating a new test

Should also test non string values such as true and object